lib/spandex_phoenix/telemetry.ex

```diff
@@ -0,0 +1,34 @@
+defmodule SpandexPhoenix.Telemetry do
+  @moduledoc """
+  Defines
+  """
+
+  @doc "Attaches `:telemetry` event handlers for Phoenix Telemetry events."
+  def attach(opts \\ []) do
+    handlers = %{
+      [:phoenix, :router_dispatch, :start] => &phoenix_router_dispatch_start/4,
+      [:phoenix, :router_dispatch, :stop] => &phoenix_router_dispatch_stop/4
+    }
+
+    tracer = Keyword.get_lazy(opts, :tracer, fn -> Application.get_env(:spandex_phoenix, :tracer) end)
+
+    for {event, handler} <- handlers, handler_id = handler_id(event) do
+      :telemetry.attach(handler_id, event, handler, %{tracer: tracer})
```

Comment:
should i add telemetry as a hard dep in mix.exs?

Reply:
I don't think we'd want to do that, but we probably want to add it an optional dependency and give a good error message if they reach this code w/o it.